node/src/components/consensus/era_supervisor.rs

```diff
@@ -711,7 +711,7 @@ where
                     .add_candidate(candidate_block, missing_evidence);
                 effects.extend(
                     self.effect_builder
-                        .validate_block(sender.clone(), proto_block)
+                        .validate_block(sender.clone(), proto_block, None)
```

Comment:
We can't do validation only in some cases: If how blocks get validated here, when observing finality ourselves, doesn't match how they get validated when we receive finalized and executed blocks from peers, the network will break.

We need to get the timestamp here, too. I think we can actually add `vv`'s timestamp to `ConsensusProtocolResult::ValidateConsensusValue` [here](https://github.com/CasperLabs/casper-node/blob/e393d61ee1c6e6cddc87dadc801d9268a1491706/node/src/components/consensus/protocols/highway.rs#L258).

Also, we have to make sure that the block proposer proposes only valid blocks, i.e. filters out all deploys with a timestamp younger than the block's.

Reply:
> Also, we have to make sure that the block proposer proposes only valid blocks, i.e. filters out all deploys with a timestamp younger than the block's.

It isn't very clear by from the path it takes through `get_chainspec()` and then GetChainspecResult, but `current_instant` is the block's timestamp from consensus. So this is already happening, but perhaps the parameters could be better named. I'll rename to `block_timestamp` to clarify.